### PR TITLE
Add Review App Feature

### PR DESCRIPTION
**Pull request by k2nr**
**Note: This PR is based on #501 and should be merged after #501**

Client changes https://github.com/degica/barcelona-cli/pull/6

### barcelona.yml

Now you can set a top-level `review` section in your `barcelona.yml`

```yaml
review:
  group: test # group name
  image_name: quay.io/degica/awesome-rails-app
  environment:
    - name: RAILS_ENV
      value: staging
    - name: DATABASE_URL
      value_from: komoju_staging/database_url # secret environment variable
  services:
    - name: web
      service_type: web
      cpu: 128
      memory: 256
      command: nginx -g "daemon off;"
```

### Review Group

Review group has the following resources which are shared with review apps that belong to this group.

- TaskRole
- BaseDomain
  - review apps domains are subdmain of its group's base domain
  - e.g. base domain `reviews.degica.com` then review app will be `branch-name.reviews.degica.com`
- Endpoint
  - All review apps that belong to a review group use the same endpoint
  - Barcelona admins need to setup DNS record so that all subdomains of group's base domain are pointed to the endpoint

```
$ bcn review group create review-group-name --endpoint endpoint-name --base-domain reviews.degica.com
```

### Review app

```
$ bcn review deploy branch-name --tag docker-image-tag
```

Once you execute the above command, your docker container will be available at `https://branch-name.reviews.degica.com`

### bcn run

```
$ bcn review list --group group-name
  SUBJECT |          DOMAIN          |                  
+---------+--------------------------+-----------------+
  v2      | v2.reviewapps.degica.com | review-7066c768  
```

`review-7066c768` is the name you want to `bcn run` into. So,

```
$ bcn run -H review-7066c768 sh
Waiting for the process to start
Connecting to the process
/ # 
```

Yay :tada:

### CODE REVIEW

**Comment by k2nr** [comment]
@essa This is ready for review. Please have look :bowing_man: 

Important review aspects are:

- We need to make sure that this change will not break existing features
- YAML specs; Is the current YAML structure good?

File: app/controllers/review_apps_controller.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
+class ReviewAppsController < ApplicationController
+  skip_before_action :authenticate, only: [:ci_create, :ci_delete]
+
+  def create
+    group = ReviewGroup.find_by!(name: params[:review_group_id])
```
Comment (essa) [inline app/controllers/review_apps_controller.rb]: How about replacing it with `before_action`?

```ruby
before_action :load_review_group

def load_review_group
  @group = ReviewGroup.find_by!(name: params[:review_group_id])
end
```

**Comment by essa** [comment]
LGTM 👍 besides my comment about a little refactoring.

**Comment by k2nr** [comment]
Thanks! I'm going to merge

**Pull request merged**